src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -100,58 +100,67 @@ public String deploy(AppDeploymentRequest request) {
 		Map<String, String> envVariables = new HashMap<>();
 
 		try {
+			Map<String, String> properties = new HashMap<>(request.getDefinition().getProperties());
+			// Remove server.port as CF assigns a port for us, and we don't want to override that
+			String port = properties.remove("server.port");
+			if (port != null) {
+				logger.info(String.format("Ignoring server.port=%s property for app %s, as Cloud Foundry will assign a dynamic port", name, port));
```

Comment:
I'd make it WARN since it is something one should ideally take action on to remove from the stream definition.